src/content/1.7/development/architecture/cqrs.md

```diff
@@ -0,0 +1,69 @@
+---
+title: CQRS
+weight: 10
+---
+
+# CQRS
+
+## What is CQRS?
+
+CQRS stands for _Command Query Responsibility Segregation_. In brief, the CQRS pattern suggests to separate "write" and "read" models, which it refers to as Commands and Queries.
+
+## Why using CQRS in PrestaShop?
+
+During Back Office migration to Symfony, PrestaShop needs a way to access and alter data on the new Symfony pages without multiplying the sources of truth, and without accessing `ObjectModel` (e.g. `Product`, `Category`) directly.
+
+Commands and Queries allow us to isolate the controllers from the data source, which can be later replaced by something else while leaving behind a nice API.
+
+This implementation proposes a "top-down" design, which inverses the classic data-driven design. It starts on a page and the actions performed in it, and then trickles down layer by layer, finishing on the data layer.
+
```

Comment:
It could be nice to have a schema here. This schema would compare old and new architecture.

In old architecture we would see the browser sending HTTP requests to a big bunch of code which would contain Controllers calling directly core classes. Like:

browser ==(send HTTP requests)==> controller ==(calls)==> core

In new architecture, we would show the clear separation of the bus

browser ==(send HTTP requests)==> SF controller =(dispatchs Queries and Commands)=> Bus routing =(Handler handles the Command/Query)=> Adapter Handler ==(calls)==> core

Reply:
Interesting, I've tried to do so in https://github.com/PrestaShop/docs/pull/232/commits/b908ef4eff4effd4a9170531bd0ebe171f3f5c5c commit, could you check?

Reply:
👍